internal/commands/exec/exec.go

```diff
@@ -156,8 +159,19 @@ func RunExec(config ExecConfig) (ret int) {
 	if config.EnvoyConfig.CertificateDirectory != "" {
 		options.Directory = config.EnvoyConfig.CertificateDirectory
 	}
+
+	tlsConfig, err := api.SetupTLSConfig(&config.ConsulConfig.TLSConfig)
```

Comment:
This is wrong, returns an unitialized `tls.Config` struct when TLS is not configured instead of `nil`, which inadvertently attempts client TLS connections when passed to `NewCertManager`